templates/vue/src/components/modals/NodeModal/forms/ContentForm/VideoForm.vue

```diff
@@ -18,68 +18,171 @@
+              :aria-label="editingKalturaId ? 'Submit' : 'Edit Kaltura ID'"
+              :disabled="isLoadingKalturaCaptions"
+              @click="handleKalturaIdEdit"
+            >
+              {{ editingKalturaId ? "Submit" : "Change" }}
+            </b-button>
+          </b-input-group-append>
+        </b-input-group>
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col>
+        <b-overlay :show="isLoadingKalturaCaptions">
+          <b-form-group label="Captions">
+            <b-form-checkbox
+              v-model="useCaptions"
+              data-qa="node-captions-toggle"
+              switch
+              @change="handleToggleCaptions"
+            >
+              {{ useCaptions ? "On" : "Off" }}
+            </b-form-checkbox>
+            <template v-if="useCaptions">
+              <caption-row
+                v-for="(caption, index) in captions"
+                :key="caption.id"
+                :value="caption"
+                :index="index"
+                :is-removable="captions.length >= 2"
+                :is-default="caption.id === defaultCaptionId"
+                :languages="languages"
+                @input="captions.splice(index, 1, $event)"
+                @setDefault="defaultCaptionId = $event"
+                @remove="removeCaption(index, caption)"
+              ></caption-row>
+              <b-button
+                class="mt-3"
+                data-qa="add-caption-button"
+                @click="addCaption"
+              >
+                <i class="fas fa-plus icon"></i>
+                Add caption
+              </b-button>
+            </template>
+          </b-form-group>
+        </b-overlay>
+        <b-form-group v-if="pendingCaptions.length > 0" label="Pending captions">
+          <caption-row
+            v-for="(caption, index) in pendingCaptions"
+            :key="caption.id"
+            :value="caption"
+            is-removable
+            is-pending
+            :languages="languages"
+            :error-message="caption.errorMessage"
+            @input="pendingCaptions[index] = $event"
+            @move="moveFromPending(index, caption)"
+            @remove="removePendingCaption(index)"
+          ></caption-row>
+        </b-form-group>
       </b-col>
     </b-row>
   </div>
 </template>
 
 <script>
+import ISO6391 from "iso-639-1"
 import FileUpload from "@/components/modals/common/FileUpload"
+import client from "@/services/TapestryAPI"
 import Helpers from "@/utils/Helpers"
 import { mapMutations } from "vuex"
+import CaptionRow from "./CaptionRow"
+
+const defaultCaption = {
+  captionUrl: "",
+  language: "English",
+  label: undefined,
+}
 
 export default {
   components: {
     FileUpload,
+    CaptionRow,
   },
   data() {
+    const captions = this.$store.state.currentEditingNode.typeData.captions ?? []
     return {
       useKaltura: false,
+      kalturaId: this.$store.state.currentEditingNode.typeData.kalturaId,
```

Comment:
I implemented it this way because the new UI allows users to edit the Kaltura ID separately before submitting it, so the Kaltura ID would only be stored into the Vuex state when "Submit" is clicked. If you have any suggestions to make this cleaner, please let me know!

Reply:
Thanks for explaining, I think it makes sense and let's keep it this way.